File: src/main/java/com/elvencompass/maps/domain/entity/Location.java
package com.elvencompass.maps.domain.entity;

import java.util.List;

import lombok.Data;

@Data
public class Location {
    private Integer id;
    private String name;
    private String description;
    private String wikiLink;
    private String mapName;
    private List<Marker> markerList;
}

File: src/main/resources/static/elvencompass/src/app/classes/location.ts
import { Marker } from "./marker"

export class Location {
    id!:number;
    name!: string|null;
    description!: string|null;
    wikiLink!: string|null;
    mapName!: string;
    markers!: Marker[];

    constructor(name:string|null, description:string|null, wikiLink:string|null){
        this.name = name;
        this.description = description;
        this.wikiLink = wikiLink;
    }
}
